src/Compilers/Core/CodeAnalysisTest/IsSymbolAccessibleWithinTests.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests
+{
+    public class IsSymbolAccessibleWithinTests
+    {
+        [Fact]
+        public void CrossLanguageException()
+        {
+            var csharpTree = CSharpSyntaxTree.ParseText("class A { }");
+            var vbTree = VisualBasicSyntaxTree.ParseText(
+@"Class A
+End Class
+");
+            var csc = CSharpCompilation.Create("CS", new[] { csharpTree }, new MetadataReference[] { TestBase.MscorlibRef });
+            var Ac = csc.GlobalNamespace.GetMembers("A")[0] as INamedTypeSymbol;
+
+            var vbc = VisualBasicCompilation.Create("VB", new[] { vbTree }, new MetadataReference[] { TestBase.MscorlibRef });
+            var Av = vbc.GlobalNamespace.GetMembers("A")[0] as INamedTypeSymbol;
+
+            Assert.Throws<ArgumentException>(() => csc.IsSymbolAccessibleWithin(Av, Av));
+            Assert.Throws<ArgumentException>(() => csc.IsSymbolAccessibleWithin(Av, Ac));
+            Assert.Throws<ArgumentException>(() => csc.IsSymbolAccessibleWithin(Ac, Av));
+            Assert.Throws<ArgumentException>(() => csc.IsSymbolAccessibleWithin(Ac, Ac, Av));
+
+            Assert.Throws<ArgumentException>(() => vbc.IsSymbolAccessibleWithin(Ac, Ac));
+            Assert.Throws<ArgumentException>(() => vbc.IsSymbolAccessibleWithin(Ac, Av));
+            Assert.Throws<ArgumentException>(() => vbc.IsSymbolAccessibleWithin(Av, Ac));
+            Assert.Throws<ArgumentException>(() => vbc.IsSymbolAccessibleWithin(Av, Av, Ac));
```

Comment:
>Assert.Throws<ArgumentException>(() => vbc.IsSymbolAccessibleWithin(Av, Av, Ac)); [](start = 12, length = 81)

It would be good to have tests with null as an invalid input.

Reply:
> It would be good to have tests with null as an invalid input.

See `AccessCheckApi_01` in `AccessCheckTests.cs`

---
In reply to: [192576121](https://github.com/dotnet/roslyn/pull/27312#discussion_r192576121) [](ancestors = 192576121)